core/src/main/java/org/elasticsearch/common/geo/GeoDistance.java

```diff
@@ -78,7 +78,7 @@ public void writeTo(StreamOutput out) throws IOException {
                     out.write(0);  // write PLANE ordinal
                     return;
                 case 1:
-                    out.write(2);  // write bwc ARC ordinal
+                    out.write(1);  // write bwc ARC ordinal
```

Comment:
@nknize I may have been wrong here, please let me know if I misunderstood the ordinal logic here!

Reply:
I am not sure I follow. As far as I can see ARC was indeed 2, and we are reading it as 2 above. Could you explain? Did it change more than once? Should we add another version check? I also think we should replace this `write` with `writeVInt` while we are at it. I mean it works for small values since at the end it's just writeByte, but it's just a bit confusing that we read one way and write another.

Reply:
> As far as I can see ARC was indeed 2

in which version? ON [5.3.0](https://github.com/elastic/elasticsearch/blob/v5.3.0/core/src/main/java/org/elasticsearch/common/geo/GeoDistance.java#L36), I only see two `GeoDistance` enums. `PLANE`(0) and `ARC`(1)

Reply:
looking further back, it seems you are right. it was `2` on [5.2.0](https://github.com/elastic/elasticsearch/blob/v5.2.0/core/src/main/java/org/elasticsearch/common/geo/GeoDistance.java#L97)

This deserves stricter bwc version checks for the different version ranges 👍